plugins/Files/js/sagas/helpers.js

```diff
@@ -177,13 +179,21 @@ export const searchFiles = (files, text, path) => {
 	let matchingFiles = List(files).filter((file) => file.siapath.indexOf(path) !== -1)
 	matchingFiles = matchingFiles.filter((file) => file.available)
 	matchingFiles = matchingFiles.filter((file) => file.siapath.toLowerCase().indexOf(text.toLowerCase()) !== -1)
-	return matchingFiles.map((file) => ({
-		size: readableFilesize(file.filesize),
-		name: Path.basename(file.siapath),
-		siapath: file.siapath,
-		available: file.available,
-		type: 'file',
-		uploadprogress: Math.floor(file.uploadprogress).toString(),
-	}))
+	return matchingFiles
 }
 
+// rangeSelect takes a file to select, a list of files, and a set of selected
+// files and returns a new set of selected files consisting of all the files
+// between the last selected file and the clicked `file`.
+export const rangeSelect = (file, files, selectedFiles) => {
```

Comment:
I'm happy with the test coverage for this function, we can always come back later and refactor it.